examples/ecs/derived_state.rs

```diff
@@ -0,0 +1,209 @@
+//! This example illustrates how to use [`DerivedStates`] for more complex state transitions.
+//!
+//! In this case, we're transitioning from a `Menu` state to an `InGame` state, with the addition
+//! of the capacity to pause and unpause if we are in game using the ESCAPE button on the keyboard.
```

Comment:
I think using `Space` to pause will be less confusing: ESCAPE usually opens the menu.